modules/mauth-signer/src/main/java/com/mdsol/mauth/SignerConfiguration.java

```diff
@@ -1,53 +1,91 @@
+import java.util.List;
 import java.util.UUID;
 
 public class SignerConfiguration implements MAuthConfiguration {
+  private static final Logger logger = LoggerFactory.getLogger(SignerConfiguration.class);
+
   public static final String APP_SECTION_HEADER = "app";
   public static final String MAUTH_SECTION_HEADER = "mauth";
   public static final String APP_UUID_PATH = APP_SECTION_HEADER + ".uuid";
   public static final String APP_PRIVATE_KEY_PATH = APP_SECTION_HEADER + ".private_key";
-  public static final String V2_ONLY_SIGN_REQUESTS = MAUTH_SECTION_HEADER + ".v2_only_sign_requests";
+  public static final String MAUTH_SIGN_VERSIONS = MAUTH_SECTION_HEADER + ".sign_versions";
 
+  public static final List<MAuthVersion> ALL_SIGN_VERSIONS = Arrays.asList(MAuthVersion.values());
+  public static final List<MAuthVersion> DEFAULT_SIGN_VERSION = Arrays.asList(MAuthVersion.MWSV2);
 
   private final UUID appUUID;
   private final transient String privateKey;
-  private boolean v2OnlySignRequests = false;
+  private List<MAuthVersion> signVersions;
 
   public SignerConfiguration(Config config) {
     this( UUID.fromString(config.getString(APP_UUID_PATH)),
         config.getString(APP_PRIVATE_KEY_PATH),
-        config.hasPath(V2_ONLY_SIGN_REQUESTS) || config.isEmpty() ? config.getBoolean(V2_ONLY_SIGN_REQUESTS) : false);
+        config.hasPath(MAUTH_SIGN_VERSIONS) ? config.getString(MAUTH_SIGN_VERSIONS) : "");
   }
 
   public SignerConfiguration(UUID appUUID, String privateKey) {
+    this(appUUID, privateKey, DEFAULT_SIGN_VERSION);
+  }
+
+  public SignerConfiguration(UUID appUUID, String privateKey, String signVersionsStr) {
     validateNotNull(appUUID, "Application UUID");
     validateNotBlank(privateKey, "Application Private key");
-
     this.appUUID = appUUID;
     this.privateKey = privateKey;
+    this.signVersions = getSignVersions(signVersionsStr);
   }
 
-  public SignerConfiguration(UUID appUUID, String privateKey, boolean v2OnlySignRequests) {
+  public SignerConfiguration(UUID appUUID, String privateKey, List<MAuthVersion> signVersions) {
     validateNotNull(appUUID, "Application UUID");
     validateNotBlank(privateKey, "Application Private key");
-
     this.appUUID = appUUID;
     this.privateKey = privateKey;
-    this.v2OnlySignRequests = v2OnlySignRequests;
+    this.signVersions = signVersions;
   }
 
-  public UUID getAppUUID() {
-    return appUUID;
-  }
+  public UUID getAppUUID() { return appUUID; }
 
   public String getPrivateKey() {
     return privateKey;
   }
 
-  public boolean isV2OnlySignRequests() {
-    return v2OnlySignRequests;
+  public List<MAuthVersion> getSignVersions() {
+    return signVersions;
+  }
+
+  static public List<MAuthVersion> getSignVersions(String signVersionsStr) {
+    List<MAuthVersion> signVersions = new ArrayList();
+    List<String> unrecognizedVersions = new ArrayList();
+    List<String> versionList = Arrays.asList(signVersionsStr.trim().toLowerCase().split(","));
```

Comment:
this trim is unnecessary if you're already trimming every individual string after splitting

Reply:
removed in https://github.com/mdsol/mauth-jvm-clients/pull/128/commits/dc293690e580e527380a15b19f4cb9e3855d4d48